### DISCUSSION TITLE
SDV values for orthotropic bamboo damage not correctly reported

### DISCUSSION THREAD

**Discussion by JAWilliams123** [thelfer/tfel#678] [Q&A] [answered]
Hello Everyone,

Apologies for the duplicate post but I believe my first post was made in the wrong location and was not able to delete the post.
I have developed the attached mfront to account for damage in bamboo based on Hill 1948 yield criteria. I was able to compile things and get it to run via Calculix, but for some strange reason, the damage state variables (dx, dy and dz) do not behave as expected. In the attached file, the damage has been hardwired to 0.1 but this is not what is shown the Calculix results. The single element CauliX input file has also been uploaded. Any assistance with this is greatly appreciated.
[BambooCalibrateDynamic.inp.txt](https://github.com/user-attachments/files/18281408/BambooCalibrateDynamic.inp.txt)
[HillYield01JW.mfront.txt](https://github.com/user-attachments/files/18281409/HillYield01JW.mfront.txt)
Many thanks and happy holidays,
JW

**Comment by thelfer** [comment]
Hi @JAWilliams123,

Using the `Implicit` DSL implies that you compute the residual and jacobian of the implicit system associated with the evolution of the state variables in the @Integrator code block. See this page for details: https://thelfer.github.io/tfel/web/implicit-dsl.html

The  `Implicit` DSL automatically declares the elastic strain as the first state variable. You use the `StandardElasticity` brick, which automatically initializes the residual associated with the elastic strain as follows:

```
feel = deel - deto;
```

See https://thelfer.github.io/tfel/web/HookeStressPotential.html for details.

In the end, what you do in the `@Integrator` code block is simply ignored, as:

1. the implicit system will lead to an elastic answer
2. the stress will be computed by the  `StandardElasticity` brick in the automatically generated `@ComputeFinalStress` code block using the Hooke law.

Your behaviour seems simple to implement (the evolution of damage seems a simple function of the strain), you may want to you the `Default` DSL as in this example: https://github.com/thelfer/tfel/blob/master/mfront/tests/behaviours/OrthotropicElastic.mfront

However, I could be of better help if I knew exactly what are exactly the constitutive equations of the behaviour you want to implement.

**Reply by JAWilliams123** [reply]
Hello @thelfer,

Happy New Year!!

Many thanks for the time taken to develop a detailed response. What I am trying to capture is the damage for a section of bamboo (anisotropic material) using the Hill yield criterion.  With the experimental damage initiation stresses given in the form of the normal stress and shear stresses in the radial and tangential directions (the global y axis is the longitudinal axis of the bamboo cylinder and the x and z axes are the tangential and radial axes respectively), I would like to adjust the stiffness matrix based on damage that occurs in any direction based on the yield criteria set out by Hill 1948. I am trying to achieve something similar to what is done in the UnilateralMazars.mfront file but accounting for damage in any of the three directions individually and all together. By separating them as statevariables, the modes of damage can be easily viewed with the stress output for the model.  
I had a go using the hillTensor functionality in MFront but had some difficulty getting things up and running. Any suggestions you have to improve the proposed MFront will be greatly appreciated. 

Best wishes,
JWilliams

**Reply by thelfer** [reply]
Based on your file, I did this. I must admit that part of the constitutive equations don't really make sense to me..

[HillYield01JW.zip](https://github.com/user-attachments/files/18295223/HillYield01JW.zip)

**Reply by JAWilliams123** [reply]
Many thank @thelfer!

I believe my damage calculation is off. Based on a linear damage model I think that it should be re-written as:
` dx = min(max_damage, beta*(eto[0] + deto[0])*E1 / fv1, real(0)));` where I have used the total strain to calculate the damage and fv1 is the damage threshold in MPa and E1 is the Young's modulus in the direction under consideration. 
Based on the updated file, I want to confirm that there is no need to update Dv to incorporate the damage that has occurred.

Many thanks,
JW

**Reply by thelfer** [reply, marked as answer]
> Based on the updated file, I want to confirm that there is no need to update Dv to incorporate the damage that has occurred.

The current implementation returns the elastic stiffness for the tangent operator which would lead to an approximate jacobian.

If you modify Dv to take damage into account, you would get the secant operator which is a better approximation.

If you want the true consistent tangent operator, you have to compute something like:

~~~~
dsig_deto = Dv + (dDv_dx * (eto + deto) ^ ddx_ddeto) + ... 
~~~~

**Reply by JAWilliams123** [reply]
Many thanks @thelfer!

Best wishes,
JW

**Reply by thelfer** [reply]
You'll thank me when it works :)
